File: domox-domain/src/main/java/domox/fixture/SimpleObject_persona.java
package domox.fixture;

import domox.dom.rqm.Author;
import domox.dom.rqm.Authors;
import lombok.AllArgsConstructor;
import org.apache.isis.applib.services.registry.ServiceRegistry;
import org.apache.isis.testing.fixtures.applib.personas.PersonaWithBuilderScript;
import org.apache.isis.testing.fixtures.applib.personas.PersonaWithFinder;
import org.apache.isis.testing.fixtures.applib.setup.PersonaEnumPersistAll;

@AllArgsConstructor
public enum SimpleObject_persona
        implements PersonaWithBuilderScript<SimpleObjectBuilder>, PersonaWithFinder<Author> {

    FOO("Foo"),
    BAR("Bar"),
    BAZ("Baz"),
    FRODO("Frodo"),
    FROYO("Froyo"),
    FIZZ("Fizz"),
    BIP("Bip"),
    BOP("Bop"),
    BANG("Bang"),
    BOO("Boo");

    private final String name;

    @Override
    public SimpleObjectBuilder builder() {
        return new SimpleObjectBuilder().setName(name);
    }

    @Override
    public Author findUsing(final ServiceRegistry serviceRegistry) {
        Authors authors = serviceRegistry.lookupService(Authors.class).orElse(null);
        return authors.findByNameExact(name);
    }

    public static class PersistAll extends PersonaEnumPersistAll<SimpleObject_persona, Author> {
        public PersistAll() {
            super(SimpleObject_persona.class);
        }
    }
}

File: module-ksimple/src/main/kotlin/org.apache.isis.domox/modules/ksimple/fixture/SimpleObject_persona.kt
package org.apache.isis.domox.modules.ksimple.fixture

import org.apache.isis.applib.services.registry.ServiceRegistry
import org.apache.isis.domox.modules.ksimple.dom.so.SimpleObject
import org.apache.isis.domox.modules.ksimple.dom.so.SimpleObjects
import org.apache.isis.testing.fixtures.applib.api.PersonaWithBuilderScript
import org.apache.isis.testing.fixtures.applib.api.PersonaWithFinder
import org.apache.isis.testing.fixtures.applib.setup.PersonaEnumPersistAll

enum class SimpleObject_persona(name: String) :
        PersonaWithBuilderScript<SimpleObjectBuilder?>,
        PersonaWithFinder<SimpleObject?> {
    FOO("Foo"),
    BAR("Bar"),
    BAZ("Baz"),
    FRODO("Frodo"),
    FROYO("Froyo"),
    FIZZ("Fizz"),
    BIP("Bip"),
    BOP("Bop"),
    BANG("Bang"),
    BOO("Boo");

    open override fun builder(): SimpleObjectBuilder? {
        return SimpleObjectBuilder().setName(name)
    }

    open override fun findUsing(serviceRegistry: ServiceRegistry): SimpleObject {
        val simpleObjects: SimpleObjects = serviceRegistry
                .lookupService(SimpleObjects::class.java)
                .orElse(null)
        return simpleObjects.findByNameExact(name)
    }

    class PersistAll : PersonaEnumPersistAll<
            SimpleObject_persona, SimpleObject?>(
            SimpleObject_persona::class.java)

}
